[fix] google images engine: Fix 'scrap_img_by_id' function

tiekoetter opened the pull request:
## What does this PR do?

The 'scrap_img_by_id' function didn't return anything useful. This fix allows the google images engine to present the full source image instead of only the thumbnail.

## How to test this PR locally?

- `make run`
- Search `!goi land`
- Check if full image is loaded in result card view

## Author's checklist

Note: If an image is redirected image proxy fails.

```
DEBUG   searx.network.image_proxy     : HTTP Request: GET https://www.pik-potsdam.de/en/institute/departments/activities/resolveuid/1921254915ba45a9a661fd8228588a90 "HTTP/1.1 301 Moved Permanently"
DEBUG   searx.webapp                  : image-proxy: wrong response code: 301
INFO    werkzeug                      : 192.168.178.129 - - [19/Feb/2022 01:37:55] "GET /image_proxy?url=https%3A%2F%2Fwww.pik-potsdam.de%2Fen%2Finstitute%2Fdepartments%2Factivities%2Fresolveuid%2F1921254915ba45a9a661fd8228588a90&h=89c65601c562be381d299b79447c42899f3d68b7939ba5691a1513822d763bf0 HTTP/1.0" 400 -
```

## Related issues

Closes #909

return42 committed:
[fix] google images engine: Fix 'scrap_img_by_id' function

The 'scrap_img_by_id' function didn't return any longer anything useful.  This
fix allows the google images engine to present the full source image instead of
only the thumbnail.

The function scrap_img_by_id() is rpelaced by a fully rewrite to parse image
URLs by a regular expression. The new function parse_urls_img_from_js(dom)
returns a mapping of data-id to image URL.

Closes: https://github.com/searxng/searxng/issues/909
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 commented on searx/engines/google_images.py:
```diff
@@ -72,11 +72,26 @@ def scrap_img_by_id(script, data_id):
     """Get full image URL by data-id in parent element"""
     img_url = ''
     _script = script.split('\n')
```
The problem of the old (and new) implementation is that there are no longer any `\n` in the result from google (the length of _split is always one).

tiekoetter replied:
So there is no need for the .split() Method? Or would you suggest another approach?

return42 replied:
> So there is no need for the .split() Method? Or would you suggest another approach?

I implemented an alternative solution in my branch [fix-909-mhe](https://github.com/return42/searxng/commits/fix-909-mhe) .. would you please have a look / see commit

tiekoetter replied:
@return42 I think your solution is much cleaner because of the regex.

How should we proceed?

return42 replied:
If it is OK for you, you can reset you branch for this PR by my branch. The alternative is I open another PR, but IMO it is better we reset this branch to have a consistent history in github's PR ..

tiekoetter replied:
sorry messed something up. will fix it now.

Done. Forgot to switch branches 🙄

return42 committed:
[fix] image_proxy: allow HTTP redirects

Without redirects the load of various images will fail when image_proxy is
enabled [1].

[1] https://github.com/searxng/searxng/pull/910#issuecomment-1045966577
Suggested-by: @dalf [1]
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 reviewed (changes requested):
@tiekoetter, @dalf: the response from google has changed, see my comment above.

I implemented an alternative solution in my branch [fix-909-mhe](https://github.com/return42/searxng/commits/fix-909-mhe) .. would you please have a look / see commit

    [fix] image_proxy: allow HTTP redirects

In this [fix-909-mhe](https://github.com/return42/searxng/commits/fix-909-mhe) branch I also implemented the redirect when image_proxy is enabled, see commit:

    [fix] image_proxy: allow HTTP redirects

return42 reviewed (approved):
Thanks!
